pallets/proof_of_existence/src/lib.rs

```diff
@@ -164,6 +185,19 @@ pub mod pallet {
                 || (values > 0 && deadline.unwrap_or_default())
         }
 
+        fn prune_old_attestations_if_needed() {
+            let newest_attestation_id = Self::next_attestation();
+            let oldest_attestation_id = Self::oldest_attestation();
+            let max_attestations = T::MaxStorageAttestations::get().saturated_into::<u64>();
+
+            if newest_attestation_id - oldest_attestation_id > (max_attestations - 1) {
+                for id in oldest_attestation_id..newest_attestation_id - (max_attestations - 1) {
+                    let _ = Values::<T>::clear_prefix(id, u32::MAX, None);
+                }
+                OldestAttestation::<T>::set(newest_attestation_id - (max_attestations - 1));
+            }
+        }
```

Comment:
Be carful on use `max_attestations - 1` When the value is `0` just works for chance. I think you need some tests about `max_attestations = 0` behaviour